spec/facility_spec.rb

```diff
@@ -23,4 +37,489 @@
+    end
+
+    it 'creates multiple facilities from Oregon (compare mapped data to API data)' do
+      facility_data = DmvDataService.new.or_dmv_office_locations
+      oregon_facilities = Facility.create_facility('OR', facility_data)
+
+      expected_data = facility_data[0]
+      expect(oregon_facilities[0].name).to eq(expected_data[:title])
+      expected_address = JSON.parse(expected_data[:location_1][:human_address]).values.join(', ')
+      expect(oregon_facilities[0].address).to eq(expected_address)
+      expect(oregon_facilities[0].phone).to eq(expected_data[:phone_number])
+      expect(oregon_facilities[0].website).to eq(expected_data[:website])
+
+      expected_data = facility_data[1]
+      expect(oregon_facilities[1].name).to eq(expected_data[:title])
+      expected_address = JSON.parse(expected_data[:location_1][:human_address]).values.join(', ')
+      expect(oregon_facilities[1].address).to eq(expected_address)
+      expect(oregon_facilities[1].phone).to eq(expected_data[:phone_number])
+      expect(oregon_facilities[1].website).to eq(expected_data[:website])
+    end
+  end
+
+  describe 'create_facility from New York' do
+    it 'fetches API data' do
+      expect(facility_data = DmvDataService.new.ny_dmv_office_locations).not_to be_nil
+    end
+
+    it 'can initialize' do
+      facility_data = DmvDataService.new.ny_dmv_office_locations
+      new_york_facilities = Facility.create_facility('NY', facility_data)
+      expected_data = facility_data[0]
+      expect(new_york_facilities[0]).to be_an_instance_of(Facility)
+    end
+
+    it 'maps data to correct data types' do
+      facility_data = DmvDataService.new.ny_dmv_office_locations
+      new_york_facilities = Facility.create_facility('NY', facility_data)
+      expected_data = facility_data[0]
+
+      expect(new_york_facilities[0].name).to be_a(String)
+      expect(new_york_facilities[0].address).to be_a(String)
+      # Certain locations in New York (kiosks) do not have phone numbers and
+      # the endpoint is just not there if that is the case.
+      expect(new_york_facilities[0].phone).to be_a(String).or be_nil
+      expect(new_york_facilities[0].website).to be_a(String)
+    end
+    it 'creates a single facility from New York (compare mapped data to API data)' do
+      facility_data = DmvDataService.new.ny_dmv_office_locations
+      new_york_facilities = Facility.create_facility('NY', facility_data)
+
+      expected_data = facility_data[0]
+      expect(new_york_facilities[0].name).to eq(Helper.title_case(expected_data[:office_name]))
+      expect(new_york_facilities[0].address).to eq("#{Helper.title_case(expected_data[:street_address_line_1])}, #{Helper.title_case(expected_data[:city])}, #{Helper.title_case(expected_data[:state])}, #{expected_data[:zip_code]}")
+      expect(new_york_facilities[0].phone).to eq(expected_data[:public_phone_number])
+      expect(new_york_facilities[0].website).to eq('undefined')
+    end
```

Comment:
Don't forget that white line space between an end and def. There is a lot in these tests so want to make it easier to read.